src/app/guest-book/components/new-message-dialog/new-message-dialog.component.spec.ts

```diff
@@ -0,0 +1,132 @@
+        provideMockStore({ initialState })
+      ]
+    }).compileComponents();
+
+    store = TestBed.inject(MockStore);
+    dialogRef = TestBed.inject(MatDialogRef) as jasmine.SpyObj<MatDialogRef<NewMessageDialogComponent>>;
+    fixture = TestBed.createComponent(NewMessageDialogComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should have form controls', () => {
+    expect(component.name).toBeTruthy();
+    expect(component.email).toBeTruthy();
+    expect(component.message).toBeTruthy();
+  });
+
+  it('save button should be disabled when invalid name provided', () => {
+    component.name.setValue('');
+    component.email.setValue(mockGuestBookMessage.author.email);
+    component.message.setValue(mockGuestBookMessage.message);
+    fixture.detectChanges();
+    const saveButton = fixture.debugElement.query(By.css('#save-button')).nativeElement;
+    expect(saveButton.disabled).toBeTruthy();
+  });
+
+  it('save button should be disabled when invalid email provided', () => {
+    component.name.setValue(mockGuestBookMessage.author.name);
+    component.email.setValue('test');
+    component.message.setValue(mockGuestBookMessage.message);
+    fixture.detectChanges();
+    const saveButton = fixture.debugElement.query(By.css('#save-button')).nativeElement;
+    expect(saveButton.disabled).toBeTruthy();
+  });
+
+  it('save button should be disabled when message with lenght smaller than 20 charactes provided', () => {
+    component.name.setValue(mockGuestBookMessage.author.name);
+    component.email.setValue(mockGuestBookMessage.author.email);
+    component.message.setValue('test short message');
+    fixture.detectChanges();
+    const saveButton = fixture.debugElement.query(By.css('#save-button')).nativeElement;
+    expect(saveButton.disabled).toBeTruthy();
+  });
+
+  it('save button should be enabled whan all valid fileds provided', () => {
+    component.name.setValue(mockGuestBookMessage.author.name);
+    component.email.setValue(mockGuestBookMessage.author.email);
+    component.message.setValue(mockGuestBookMessage.message);
+    fixture.detectChanges();
+    const saveButton = fixture.debugElement.query(By.css('#save-button')).nativeElement;
+    expect(saveButton.disabled).toBeFalse();
+  });
+
+  it('should close the dialog when dismiss is called', () => {
+    component.dismiss();
+    expect(dialogRef.close).toHaveBeenCalled();
+  });
+
+  it('should dispatch createGuestBookMessage action on save', () => {
+    const spy = spyOn(store, 'dispatch');
+    component.name.setValue(mockGuestBookMessage.author.name);
+    component.email.setValue(mockGuestBookMessage.author.email);
+    component.message.setValue(mockGuestBookMessage.message);
+    component.save();
+
+    expect(spy).toHaveBeenCalledWith(GuestBookActions.createGuestBookMessage({
+      id: jasmine.any(String) as unknown as string,
```

Comment:
casting looks unnecessary - what error says?

Reply:
@kaspars01 If both casts removed id: jasmine.any(String), i have error:
Type 'AsymmetricMatcher<any>' is not assignable to type 'string'.
In case id: jasmine.any(String) as string, error is:
Conversion of type 'AsymmetricMatcher<any>' to type 'string' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.